### ISSUE TITLE
El Capitan recording issues

### ISSUE THREAD

**Issue by Wouter1** [Wouter1/EMU-driver#65]
**IMPORTANT** please comment on this ticket only if you have something new to report regarding recording with my driver on El Capitan

In #64 @vitalibondar reported

Recorded sound is distorted (Reaper, Logic, Quicktime).
https://cloudup.com/ihcV-dV6lWL

**Comment by Wouter1** [comment]
@vitalibondar do you see any problem reports in the log files while recording?

**Comment by vitalibondar** [comment]
@Wouter1 Yes. http://pastebin.com/gyAkjkNp

And another video with clean driver installation (update after installation native original driver). I see error after while: https://cloudup.com/iV7M_CWCNpU

<img width="1280" alt="screen shot 2015-10-03 at 12 12 02" src="https://cloud.githubusercontent.com/assets/10779590/10262331/6c10071e-69cc-11e5-80bd-7477b7dab6d1.png">

Just for information: original EMU driver installed without errors, but don't work (no playback, no recording, no messages in kernel log).

**Comment by vitalibondar** [comment]
Many users are reporting problems with different usb-audio interfaces in El Capitan

**Comment by Wouter1** [comment]
ok thanks vitalbondar for the log contents.


Basically the log contains this
```
Oct 3, 2015, 12:13:42 kernel[0]: RingBufferDefault<USBInputRing>::push warning. Ignoring overrun
Oct 3, 2015, 12:13:42 kernel[0]: EMUUSBAudioEngine::convertInputSamples READ HICKUP
Oct 3, 2015, 12:13:42 kernel[0]: EMUUSBAudioEngine::convertInputSamples err reading ring: e00002e7
Oct 3, 2015, 12:13:42 kernel[0]: EMUUSBAudioEngine::convertInputSamples READ HICKUP
Oct 3, 2015, 12:13:42 kernel[0]: EMUUSBAudioEngine::convertInputSamples err reading ring: e00002e7
Oct 3, 2015, 12:13:42 kernel[0]: EMUUSBAudioEngine::convertInputSamples err reading ring: e00002e7
Oct 3, 2015, 12:13:42 kernel[0]: EMUUSBAudioEngine::convertInputSamples READ HICKUP
Oct 3, 2015, 12:13:42 kernel[0]: EMUUSBAudioEngine::convertInputSamples READ HICKUP
Oct 3, 2015, 12:13:42 kernel[0]: EMUUSBAudioEngine::convertInputSamples READ HICKUP
Oct 3, 2015, 12:13:42 kernel[0]: EMUUSBAudioEngine::convertInputSamples READ HICKUP
Oct 3, 2015, 12:13:42 kernel[0]: EMUUSBAudioEngine::convertInputSamples READ HICKUP
Oct 3, 2015, 12:13:42 kernel[0]: EMUUSBAudioEngine::convertInputSamples READ HICKUP
Oct 3, 2015, 12:13:42 kernel[0]: RingBufferDefault<USBInputRing>::push warning. Ignoring overrun
Oct 3, 2015, 12:13:42 kernel[0]: EMUUSBAudioEngine::convertInputSamples READ HICKUP
Oct 3, 2015, 12:13:42 kernel[0]: EMUUSBAudioEngine::convertInputSamples err reading ring: e00002e7
Oct 3, 2015, 12:13:42 kernel[0]: EMUUSBAudioEngine::convertInputSamples READ HICKUP
Oct 3, 2015, 12:13:42 kernel[0]: EMUUSBAudioEngine::convertInputSamples err reading ring: e00002e7
Oct 3, 2015, 12:13:42 kernel[0]: EMUUSBAudioEngine::convertInputSamples err reading ring: e00002e7
Oct 3, 2015, 12:13:42 kernel[0]: EMUUSBAudioEngine::convertInputSamples READ HICKUP
Oct 3, 2015, 12:13:42 kernel[0]: EMUUSBAudioEngine::convertInputSamples READ HICKUP
```

**Comment by Wouter1** [comment]
It seems that the driver works fine for some but not for others. Normally, recording has to work first, after that playback (which is sync'ed to the recording channel).

@djohayn you reported playback works fine on your machine for playback. What machine  do you have (hardware)?

@vitalibondar what is your machine?

**Comment by Wouter1** [comment]
0x2e7 = data underrun. We are not receiving the expected amount of data from the EMU.

@vitalibondar do you have the EMU directly connected to your computer or is there a hub in between?

**Comment by Wouter1** [comment]
>Just for information: original EMU driver installed without errors, but don't work (no playback, no recording, no messages in kernel log).

Thanks.

Not sure what to make of that. But it suggests that something important in the USB or audio system has changed. 

The original driver has the original code signature from creative so that explains why it loads without the extra hacks.

**Comment by Wouter1** [comment]
My first impression is that the USB system calls back my code with a signal that the data is ready, but that the data is not there.

**Comment by Wouter1** [comment]
@vitalibondar I opened your linked audio sample  https://cloudup.com/ihcV-dV6lWL with audacity and I see you have four channels sampled.

Can you try what it does if you sample only 2 channels, eg with audacity?

**Comment by Wouter1** [comment]
Checked the code that generates the reported error messages.

This happens when CoreAudio (the mac kernel) calls our code to fetch audio samples but too early, when the samples are not available. 

This can mean that the coreaudio is out of sync, or that there is not enough data coming in from the USB. Or maybe something does not initialize properly.

**Comment by vitalibondar** [comment]
@Wouter1 3 files: mono (reaper), stereo (audacity), 4 channels (quicktime) https://cloudup.com/cAPCLc1UF-Y
Something else: after turning on/turning off EMU, driver cannot loaded. I must reinstall it.

**Comment by vitalibondar** [comment]
@Wouter1 I have MacBook Pro (Retina, 13-inch, Late 2013) 8 GB RAM.

**Comment by Wouter1** [comment]
Thanks for the update. This should all work fine on a late 2013 rMBP, I even can use the rMBP with a hub between the compu and the EMU.

 I hear 'one two' very soft, with loads of echo in the reaper mono. 
The stereo has just loads of clicking. 
The 4 channels has same 'one two', again very soft and only on channel 1. 

There are repeated blocks of silense in the recording, and duplicate blocks of data.

**Comment by Wouter1** [comment]
As you mentioned, there seem lots of troubles with audio in El Capitan.

Maybe we should wait a bit and see how this develops after the first update of El Capitan?

**Comment by vitalibondar** [comment]
@Wouter1 I agree, that it makes sense.

**Comment by Wouter1** [comment]
EmptyFruit also has a rMBP, just a bit newer. 

Some things have changed on the audio side : dual microphones, and support for digital out through the headphone port (I suppose). But on first thought, that would only complicate matters on the NEW rMBP, not on the old one where the drivers were written for. 

I will wait till El Capitan 2 is out before digging deeper into this.

**Comment by Wouter1** [comment]
@vitalibondar just a small shot in the dark, but can you confirm that your /System/Library/Extensions/EMUUSBAudio.kext (and all files inside) have mode 755 (drwxr-xr-x), and are owned by "root", group "wheel"?

**Comment by Wouter1** [comment]
And, what does this give you
```
kextutil -tn /System/Library/Extensions/EMUUSBAudio.kext/
```

**Comment by vitalibondar** [comment]
> @vitalibondar just a small shot in the dark, but can you confirm that your /System/Library/Extensions/EMUUSBAudio.kext (and all files inside) have mode 755 (drwxr-xr-x), and are owned by "root", group "wheel"?

@Wouter1 Sorry, but i can't check this. I have installed Yosemite.

**Comment by Wouter1** [comment]
ok thanks, I hoped you had El Capitan installed on a separate partition.

**Comment by vitalibondar** [comment]
@Wouter1 I'm sorry, but I just updated my computer when El Capitan has appeared in App Store.

**Comment by Philipondio** [comment]
Hi Wouter & Vitali,
so I'm wondering: If I don't use the interface to record, but solely to listen to music, will that work on El Capitan?
I have the Emu Tracker Pre USB btw., and no problems with my other interfaces (RME Babyface, Geekout 1000).
Thanks! Phil

**Comment by Wouter1** [comment]
Thanks for your interest in my driver.

I received mixed reports about how it works in El Capitan. For most users it seems fine (there were 135 visitors last week  and I assume most were here to download it). Vitalibondar however never got it working and had to go back to Yosemite. 

I did not yet switch to El Capitan (I wait at least till version 2 I think) so I have no first hand feedback.

If you try, can you please report back here if it works for you.

**Comment by Wouter1** [comment]
I got a thank-you message from Joachim from Germany for a nicely working EMU tracker pre under mac os el capitan.

Just mentioning as this is another indication that for most users the driver still works ok.

**Comment by digdelu** [comment]
Record audio (EMU-0404 USB) not work on OS X El Capitan. 
Direct Monitor works fine!
Midi IN/OUT without complaints!
S/PDIF not tested.

 MacBook Pro Retina, 15-inch, Mid 2014

**Comment by Wouter1** [comment]
@digdelu  thanks for reporting! You're also the first to report that midi is working, great to hear that.

**Comment by Wouter1** [comment]
I created an external drive with a comletely fresh install of El Capitan to test this issue.

Unfortunately El Capitan was completely crashing every time I plugged in the network cable. This ate up all my time. Filed a bug report with Apple 23132686

**Comment by Emptyfruit** [comment]
Got time to test recording with El Capitan. More precisely:
El Capitan 10.10.11.
MBPr mid 2015, Logic x.
kext signing - disabled.

Midi: works fine.
playback: woks fine in system (music, web, etc.), seems to work fine in DAW, but get some instability. had to restart the device couple of times, especially after settings change.
recording: sound is distorted. It is not clicking, it is distortion of all input audio signal.

More interesting things: i wanted to completely remove the driver (as is was installed before updating to 11.1) and install it again, but permissions to execute commands, provided in Uninstall guide are denied. Even with SIP fully disabled.

**Comment by Wouter1** [comment]
@Emptyfruit 
Yes I'm aware of the recording issues. Im doing more detailed work on it in #70 but so far I could not find the source of the issue. 

Playback might become unstable if there are excessive amount of logs while playing. I noticed this particularly with Quicktime latest version. It seems a bug in Quicktime. Can you check your system.log during playback to see if that is the cause of the instability.

Regarding the uninstalling: thanks for reporting. I think  that you need to put SUDO in front of them. Normal users have no permissions there. I will update the uninstall docu page.

Can you retry with the fixed commands?

**Comment by Emptyfruit** [comment]
@Wouter1 Yeap, SUDO helped. I reinstalled driver. recording issue is the same, sadly. Midi worked only after rebooting.

**Comment by Wouter1** [comment]
Yes recording in El Capitan is broken, rebooting wont help. I will put that on the main page for now..

**Comment by Emptyfruit** [comment]
Do you think there's hope? :)

**Comment by Wouter1** [comment]
If you need the driver now, stick with Yosemite or Mavericks. 

I only have so an hour or so in some of the evenings, and only 1 machine so debugging is a pain. The driver was never known for its clean code either (it's a fixed relic from Creative)... Also I need some time to answer requests from users and update pages.

Main work on the issue is reported in #70.

**Comment by Emptyfruit** [comment]
I was new to OS X. Updated to 10.11 without backing up.. there is no going back for me :( I wish you luck. And more free time :)

**Comment by Wouter1** [comment]
I think they have to fix at least 2 fatal issues with Capitan before I can continue working on this.

First, it hangs up if you plug in the network cable radar issue 23223881

Second, it hangs randomly way too often. radar issue 23132686

Got no feedback from Apple on it. I have already lost 3 evenings just filing bug reports and waiting for systems to come alive again and for the moment I'm done with El Capitan.

**Comment by Wouter1** [comment]
Just an update for those hoping that I can work on the issue.

I saw that Apple released El Capitan update 10.11.1

I tried to install it on an external USB drive, but the updater crashed/hang up. After that, El Capitan did not boot at all anymore. 

I tried to re install El Capitan on my USB drive. But the installer crashes at 16%

I did a long (half an hour) hardware test of my compu in Mavericks. It ran for half an hour without any reported problem. 

I'm glad that I did this all on the external USB drive, but that's about all the news I have so far. I'm waiting for Apple to respond on the issues I reported (radar  23132686, 23223881, 23339590)

**Comment by adrian-polglase** [comment]
I would be so happy if this could be fixed, because I am experiencing the same problem in El Capitan - it simply is impossible to record without the audio being broken and heavily distorted - basically unrecognisable.

**Comment by Wouter1** [comment]
Sorry, El Capitan does not even install on my computer since their latest update. Apple has to do its job before I can do anything. I have several bug reports on this open with Apple.

**Comment by Wouter1** [comment]
Update: Finally I managed to get Capitan installed on my machine. I think I can make some time to have a look at the issues.

**Comment by adrian-polglase** [comment]
This would be amazing. Thank you Wouter1!

**Comment by Emptyfruit** [comment]
I wish you luck, man :)

**Comment by Wouter1** [comment]
@adrian-polglase 
@Emptyfruit 

The source of the issue has been localized as broken low-latency support in Apple's USB driver.

That allowed me to make a high latency workaround:

if you set the driver latency to 35ms  and the DAW to 5ms then it seems to work.

The driver latency can be set to 35 ms as follows

1. Download a fresh installer from GIT
2. Edit the lowLatency script in your favourite text editor
3. Change the 1000 into 35000
4. save the script
5. run the lowLatency script
6. run the kextInstall script
7. wait a few minutes (OSX needs time to rebuild the kernel)
8. reboot (should not be necessary but I found sometimes the updated kext does not load in Capitan).

**Comment by Wouter1** [comment]
@adrian-polglase 
@Emptyfruit 

can you please confirm if this solution works for you? THen I can announce that we at least have a workaround until Apple responds on the USB issue.

**Comment by adrian-polglase** [comment]
Thanks Wouter1. I'll give it a try now and get back to you.

**Comment by Emptyfruit** [comment]
Wow, that good news. Unfortunately, i don't have a time machine backup right now, so can't risk going to El Capitan again to test. I'll join the "test team" as soon as i can :))

**Comment by Wouter1** [comment]
@Emptyfruit  No problem. Again, this is a high-latency workaround, not a real fix. If you need that EMU I would wait for a good fix.

**Comment by Wouter1** [comment]
@vitalibondar 
@digdelu 
@Philipondio 

Above you can find a high-latency workaround. Please inform me about results if you try it

**Comment by adrian-polglase** [comment]
I can confirm that on my El Capitan machine at least, this workaround seems to work for now and allow me to once again record

Thanks Wouter1!

**Comment by Wouter1** [comment]
@adrian-polglase  Thanks for the confirmation!

**Comment by boredomisacrime** [comment]
Hi! first thanks for trying to get this to work, i started reading this because i have the same issue as vitalibondar...I have a 15 inch macbook pro with 16 gb ram running el capitan and an 0404 usb...dunno if that info is worth crap, but anyway, after around 10 seconds I get the info that the .kext file cannot be used, even after your last suggestion it still gives me the same error ... any suggestions?

**Comment by Wouter1** [comment]
@boredomisacrime  sorry for late response, I was busy working on the recording issue and did not look again here.

Please post relevant log info that says anything about the EMU kext.

Is your machine still in kext-develop mode?

**Comment by boredomisacrime** [comment]
@Wouter1  no worries, still awesome that you are trying to get this to work.
Regarding the issues, this is what i got from Terminal
![img_4439](https://cloud.githubusercontent.com/assets/16087872/11529095/aeacb672-98e2-11e5-9eff-9a3e324cea09.jpg)

**Comment by Wouter1** [comment]
Yes that seems fine. I was just wondering if it somehow got re-enabled after rebooting.

Furthermore, do you see anything concerning the EMU kext in system.log when you reboot and when the kext stops working?

**Comment by boredomisacrime** [comment]
so this is what the system.log had to say about kext :
Dec  2 12:13:25 Eduardos-MBP kernel[0]: Kext loading now disabled.
Dec  2 12:13:25 Eduardos-MBP kernel[0]: Kext unloading now disabled.
Dec  2 12:13:25 Eduardos-MBP kernel[0]: Kext autounloading now disabled.
Dec  2 12:16:29 localhost kernel[0]: kext submap [0x<ptr> - 0x<ptr>], kernel text [0x<ptr> - 0x<ptr>]
Dec  2 12:16:29 localhost kernel[0]: AirPort_Brcm4360::init AirPortFamily_kexts-1101.24 "AirPortFamily_kexts-1101.24" Sep 24 2015 19:56:07
Dec  2 12:16:39 Eduardos-MBP com.apple.xpc.launchd[1] (com.apple.dockextra.server): The HideUntilCheckIn property is an architectural performance issue. Please transition away from it.
Dec  2 12:27:28 Eduardos-MBP sudo[733]:       Ed : TTY=ttys000 ; PWD=/Users/Ed/Downloads/EMU-driver-master ; USER=root ; COMMAND=/sbin/kextunload /System/Library/Extensions/EMUUSBAudio.kext
Dec  2 12:27:28 Eduardos-MBP kernel[0]: Kext com.emu.driver.EMUUSBAudio not found for unload request.
Dec  2 12:27:29 Eduardos-MBP sudo[738]:       Ed : TTY=ttys000 ; PWD=/Users/Ed/Downloads/EMU-driver-master ; USER=root ; COMMAND=/bin/rm -r /System/Library/Extensions/EMUUSBAudio.kext
Dec  2 12:27:29 Eduardos-MBP sudo[744]:       Ed : TTY=ttys000 ; PWD=/Users/Ed/Downloads/EMU-driver-master ; USER=root ; COMMAND=/bin/cp -r EMUUSBAudio.kext /System/Library/Extensions
Dec  2 12:27:29 Eduardos-MBP sudo[754]:       Ed : TTY=ttys000 ; PWD=/System/Library/Extensions ; USER=root ; COMMAND=/usr/bin/chgrp -R wheel EMUUSBAudio.kext
Dec  2 12:27:29 Eduardos-MBP sudo[756]:       Ed : TTY=ttys000 ; PWD=/System/Library/Extensions ; USER=root ; COMMAND=/sbin/kextload -v EMUUSBAudio.kext
Dec  2 12:27:29 Eduardos-MBP com.apple.kextd[44]: Cache file /System/Library/Caches/com.apple.kext.caches/Directories/System/Library/Extensions/KextIdentifiers.plist.gz is out of date; not using.
Dec  2 12:27:29 Eduardos-MBP com.apple.kextd[44]: kext-dev-mode allowing invalid signature -67062 0xFFFFFFFFFFFEFA0A for kext "/System/Library/Extensions/EMUUSBAudio.kext"
Dec  2 12:27:34 Eduardos-MBP com.apple.kextd[44]: Cache file /System/Library/Caches/com.apple.kext.caches/Startup/KextPropertyValues_OSBundleHelper_x86_64.plist.gz is out of date; not using.
Dec  2 12:27:34 Eduardos-MBP com.apple.kextd[44]: Rescanning kernel extensions.
Dec  2 12:27:34 Eduardos-MBP com.apple.kextd[44]: Cache file /System/Library/Caches/com.apple.kext.caches/Startup/IOKitPersonalities_x86_64.ioplist.gz is out of date; not using.
Dec  2 12:27:35 Eduardos-MBP com.apple.kextcache[773]: rebuilding //System/Library/PrelinkedKernels/prelinkedkernel
Dec  2 12:27:40 Eduardos-MBP com.apple.kextcache[776]: / locked; waiting for lock.
Dec  2 12:27:46 Eduardos-MBP com.apple.kextcache[774]: kext-dev-mode allowing invalid signature -67062 0xFFFFFFFFFFFEFA0A for kext EMUUSBAudio.kext
Dec  2 12:28:04 Eduardos-MBP com.apple.kextcache[774]: Created old kernelcache copy "/System/Library/Caches/com.apple.kext.caches/Startup/kernelcache"
Dec  2 12:28:04 Eduardos-MBP com.apple.kextcache[774]: Created prelinked kernel "/System/Library/PrelinkedKernels/prelinkedkernel"
Dec  2 12:28:04 Eduardos-MBP com.apple.kextcache[774]: Created prelinked kernel using "/System/Library/Kernels/kernel"
Dec  2 12:28:04 Eduardos-MBP com.apple.kextcache[773]: /System/Library/PrelinkedKernels/prelinkedkernel not cached.
Dec  2 12:28:05 Eduardos-MBP com.apple.kextcache[773]: Successfully updated disk0s3.
Dec  2 12:28:05 Eduardos-MBP com.apple.kextcache[776]: Lock acquired; proceeding.
Dec  2 12:28:05 Eduardos-MBP com.apple.kextcache[776]: /: helper partitions appear up to date.

**Comment by Wouter1** [comment]
The log says that the EMU kext was loaded succesfully. 
It should work ok after that point. 

Do you still have the issue?

**Comment by boredomisacrime** [comment]
Yup the device doesn't show up on any software...but it can be human error mind you, i'm gifted that way :), right now i don't have the 0404 next to me, but i'll try it later today and report back.

**Comment by boredomisacrime** [comment]
Ok, so EMU now works, in the sense that sound comes out. But input is laggy and all chopped up...but please help others first, as I got a line 6 pod x3 live that will tide me over until the record solution is fixed. Sorry to have wasted your time on a non issue

**Comment by Wouter1** [comment]
@boredomisacrime 
Ok that's good to hear and what was expected at this point. 
Can you try the high-latency workaround I described above?

**Comment by boredomisacrime** [comment]
I did, but no luck. Still chunky as hell. I had it connected through a hub, then changed to direct connection and heard no difference...

**Comment by Wouter1** [comment]
Did you use a fresh downloaded driver when following the high-latency instructions? The lowlatency script can be used only once on a fresh copy.

**Comment by maxflex** [comment]
WORKING GREAT ON EL CAPITAN NOW! I'm so happy, I've been unable to record since the last OS update. Thank you so much!

**Comment by maxflex** [comment]
@Wouter1 *The lowlatency script can be used only once*

Only once? What if I want to reset the latency? 35000 was too high for me and I feel a little bit of delay during live recording.

**Comment by Wouter1** [comment]
@maxflex I just upgraded to El Capitan 10.11.2 and tested. 

However the issue has not yet been fixed. From the errors it looks like the driver is still NOT receiving low latency USB data. 

Apple did not even acknowledge my bug report yet.

If you want to re-install the driver but now without latency modifications, re-downloading the driver fresh from git is the recommended way.

**Comment by xanview** [comment]
Still the same with 10.11.4 :( - but the high latency workaround works!

**Comment by Wouter1** [comment]
@xanview  Thanks for the update. Unfortunately Apple is completely ignoring this issue.

**Comment by Laximus** [comment]
Hello, I installed the high latency workaround and it works on 10.11.4, with EMU USB 0202, THANK YOU! =)
One strange thing, when I record audio, the signal is recorded SOONER, around 1/64 in Logic ProX. So its like a positive delay - sorry if I express not correct, I am Hungarian. 
I/O Buffer size is 32 samples, with this high lateny it makes 144,6ms roundtrip latency. How could it be recorded sooner? =) 
One more question: if I would like to back the normal latency, do I have to rewrite the 35000 to 1000 again in the lowlatency script? Thank you.

**Comment by roddy20** [comment]
<key>SafetyOffsetMicroSec</key>
			<integer>4200</integer>  is NORMAL
and
<key>SafetyOffsetMicroSec</key>
			<integer>1000</integer> is LOW
You can write this in Textedit with no script required

**Comment by Laximus** [comment]
Thank you! And where is the file in the system to edit the numbers?

**Comment by roddy20** [comment]
01 Apr 2016, в 15:00, Laximus <notifications@github.com> написал(а):

> Thank you! And where is the file in the system to edit the numbers?
> 
> 
> 

Hello!
You can edit inside the EMU**kext info.plist and reinstall it again

**Comment by Laximus** [comment]
Thank you

**Comment by Laximus** [comment]
I noticed with the high latency workaround, if the iMac goes to sleep for longer period, and when I want to use it again, it is just freezes with a half dark screen, and i have to reset the computer. It start with a screen like this, but after everithing is ok until the next sleep...

![kernel-panic-ml_300-100024406-medium](https://cloud.githubusercontent.com/assets/18173223/14225182/cabee9c0-f8b9-11e5-8fc2-f7cae11b22c9.png)

Does anyone has the same experience?

I have this configuration:

![kepernyofoto 2016-04-02 - 10 03 45](https://cloud.githubusercontent.com/assets/18173223/14225197/43fd9ef8-f8ba-11e5-9d67-6e24c890c793.jpg)

**Comment by Wouter1** [comment]
@roddy20  @Laximus  Sorry for the late response, I am on holidays and mostly without internet. Yes you can change the plist manually and re-install, or just re-download and install the driver from github.

I am not aware of the issue with sleeping for longer periods.

**Comment by Laximus** [comment]
@Wouter1 Thank you the answer. =)  Yes it's strange , but its only happen when  the high latency workaround installed. When I revert to the original latency, there is no this sleeping issue. Thank you anyway.

**Comment by Laximus** [comment]
@Wouter1 One more question please.: first when I installed your driver I had to enter this command: csrutil enable --without kext in terminel in recovery mode. Do I have to do it every time when I am changing the latency time or revert to the origonal latency or not? Or do I have to do anything in recovery to change this line back to the original state? Hope you understand what I mean. Thank you!

**Comment by Wouter1** [comment]
@Laximus  the csrutil setting is remembered so doing it once is sufficient. Maybe iyou need to re-run if you upgrade to a higher version of the OS. 

It has nothing to do with the latency of the driver.

To recover to original state, just re-download the driver. The low-latency script changes values in the original file and can be used only once.

**Comment by Laximus** [comment]
@Wouter1 Thank you =)  I mean the original state of the csrutil.

**Comment by Wouter1** [comment]
@Laximus 
oh, for the original state of csrutil, you have to re-run csrutil I assume. Never tried that. You will need to check the csrutil to disable the option. I expect something like csrutil disable 

 Notice that you must leave it enabled (so --without kext) if you want to continue using the driver.

**Comment by Laximus** [comment]
@Wouter1 Thank you the quick response. There was an Os update, and after the driver works like charm, I wouldnt have to do it again. =) But I did.  =DDDD   It works.

**Comment by Emptyfruit** [comment]
@Laximus , hi. Can you, please, be more specific, what update and what you mean by "works like a charm" ?

**Comment by Laximus** [comment]
@Emptyfruit Hi! Of course. There was an OSX update around a week ago: 10.11.4 and after this update the EMU-driver from here and the EMU device works great, and  the high latency workaround works too for audio recording. Sorry for not to be clear.  =) My device is an EMU USB 0202.

**Comment by Emptyfruit** [comment]
Thanks. So the recording issue is still present. :(

**Comment by eabarth** [comment]
This driver worked well for me the first time I used it. I plugged in a mic and it sounded totally normal. Went to use it today, and the recording is all garbled. Weird that it just stopped working all of a sudden. I haven't updated anything.

**Comment by Wouter1** [comment]
@eabarth  are you on El Capitan and did you apply the high latency setting?

**Comment by eabarth** [comment]
Thanks for responding so fast! I'm on El Capitan (latest update) but haven't messed around with latency settings. Since it just worked the first time I tried it, I thought I'd be good with the defaults.

I don't see instructions on how to apply a high latency setting, just the article on how to make it low latency. If there's a common fix I should be trying, can you link me to instructions? Sorry to bother you with my noobishness.

**Comment by Wouter1** [comment]
@eabarth I updated the installation pages and added an extra script to simplify installation on El Capitan.
Can you give it a try?

**Comment by eabarth** [comment]
Yes, I got it to work! Thanks for that!

**Comment by eabarth** [comment]
Ugh, nope. It worked for a while when I had a direct line plugged in, but when I switched over to my condenser mic (which needs phantom power), it suddenly became all garbed again, and now even the direct line sounds garbled.

The reason I specifically mention the condenser mic is because my dynamic mic worked just fine (that's what I used after installing this driver the first time), and only yesterday did I try my condenser mic, which is when the garbled recording started. 

I imagine I will have to uninstall and reinstall the driver to get it back to how it was working, but I would really like to be able to use my condenser mic if possible. Do you know of any reason the condenser mic, or maybe turning on phantom power, would cause all recording to become garbled from then on?

**Comment by Wouter1** [comment]
@eabarth  glad you got it working. Yes sometimes the interface needs to be resete with the control panel. My guess is that the EMU settings sometimes get garbled.

**Comment by Laximus** [comment]
@Wouter1 Hello, I would like to write a report, that the wake up issues on El Capitan/iMac somehow is related to this driver or to the hardware itself. Or the Apple didnt write properly something in the USB drivers. I used this driver, and was lot of wake up issues with EMU USB 0202 -  kernel panics, and have to restart the iMac almost from every second sleep state. Around a week ago I dont use the EMU USB 0202 at all, not even pluged by usb, and no wake up issues at all. I really apreciate your work of course, I just want to report this. Thank you.

**Comment by Wouter1** [comment]
@Laximus thanks for reporting. I madee separate ticket as this is not a recording issue.

**Comment by Wouter1** [comment]
ANYONE please comment on this ticket only if you have something new to report regarding recording with my driver on El Capitan

**Comment by Laximus** [comment]
Oh, it's true. Sorry for the OFF!

**Comment by Wouter1** [comment]
@vitalibondar  @Philipondio  @digdelu  @eabarth @Emptyfruit @adrian-polglase @boredomisacrime  @maxflex  @Laximus @roddy20

Good news! 

New version of the EMU driver! The latency and recording issues seem to be resolved. 
Can you please check if it works for you?

Sorry for the long time this took. It took a huge amount of work to adapt all the changes that Apple made to the USB drivers.

**Comment by Emptyfruit** [comment]
Wow, that's very good news. I hope I'll be able to try shortly. Also let's
hope apple won't kill it again in Sierra.. :)

**Comment by Laximus** [comment]
WORKS! WORKS! Thank you so much! =) I tried it with the low latency version, audio record is perfect. I will try later with normal latency as well! Yeehaaa!  Thank you!!!!

**Comment by Wouter1** [comment]
Great to hear! Glad that it all works :)

**Comment by sookiesookie6** [comment]
This works for me except that a lot of plugins no longer work, when I switch back to built in audio they start working again. Just talking about built in plugins like the Pitch plugins etc.

Is there a setting I need to change perhaps?
